### ISSUE TITLE
Error Creating Stream with TIMESTAMP (property) Defined on TIMESTAMP (data type) Column.

### ISSUE THREAD

**Issue by albp-rmt** [confluentinc/ksql#8937]
**Describe the bug**

I get the following error when I try to override the ROWTIME pseudo column, with a TIMESTAMP column (defined as timestamp-millis in avro) to perform a windowing operation: -

```
Timestamp column, `TS`, should be LONG(INT64) or a String with a timestamp_format specified
```

Whilst my actual scenario is more complicated, I have managed to reproduce the problem by following the simple example in the KsqlDB documentation [How to use a custom timestamp column](https://docs.ksqldb.io/en/latest/how-to-guides/use-a-custom-timestamp-column/#timestamps-represented-by-timestamp-columns).

**To Reproduce**

```
ksql> CREATE STREAM s5 (
    k VARCHAR KEY,
    ts TIMESTAMP,
    v1 INT,
    v2 VARCHAR
) WITH (
    kafka_topic = 's5',
    partitions = 1,
    value_format = 'avro',
    timestamp = 'ts'
);


Timestamp column, `TS`, should be LONG(INT64) or a String with a timestamp_format specified
```

The statement also produces this in the logs: -

```
io.confluent.ksql.util.KsqlStatementException: Timestamp column, `TS`, should be LONG(INT64) or a String with a timestamp_format specified
Statement: CREATE STREAM S5 (K STRING KEY, TS TIMESTAMP, V1 INTEGER, V2 STRING) WITH (KAFKA_TOPIC='s5', KEY_FORMAT='KAFKA', PARTITIONS=1, TIMESTAMP='ts', VALUE_FORMAT='AVRO');
	at io.confluent.ksql.engine.EngineExecutor.plan(EngineExecutor.java:405)
	at io.confluent.ksql.engine.SandboxedExecutionContext.plan(SandboxedExecutionContext.java:127)
	at io.confluent.ksql.rest.server.computation.ValidatedCommandFactory.createForPlannedQuery(ValidatedCommandFactory.java:134)
	at io.confluent.ksql.rest.server.computation.ValidatedCommandFactory.createCommand(ValidatedCommandFactory.java:106)
	at io.confluent.ksql.rest.server.computation.ValidatedCommandFactory.create(ValidatedCommandFactory.java:66)
	at io.confluent.ksql.rest.server.validation.RequestValidator.validate(RequestValidator.java:164)
	at io.confluent.ksql.rest.server.validation.RequestValidator.validate(RequestValidator.java:130)
	at io.confluent.ksql.rest.server.resources.KsqlResource.handleKsqlStatements(KsqlResource.java:281)
	at io.confluent.ksql.rest.server.KsqlServerEndpoints.lambda$executeKsqlRequest$2(KsqlServerEndpoints.java:204)
	at io.confluent.ksql.rest.server.KsqlServerEndpoints.lambda$executeOldApiEndpointOnWorker$23(KsqlServerEndpoints.java:358)
	at io.confluent.ksql.rest.server.KsqlServerEndpoints.lambda$executeOnWorker$22(KsqlServerEndpoints.java:344)
	at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$2(ContextImpl.java:313)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: io.confluent.ksql.util.KsqlException: Timestamp column, `TS`, should be LONG(INT64) or a String with a timestamp_format specified
	at io.confluent.ksql.execution.streams.timestamp.TimestampExtractionPolicyFactory.create(TimestampExtractionPolicyFactory.java:86)
	at io.confluent.ksql.execution.streams.timestamp.TimestampExtractionPolicyFactory.validateTimestampColumn(TimestampExtractionPolicyFactory.java:42)
	at io.confluent.ksql.ddl.commands.CreateSourceFactory.buildTimestampColumn(CreateSourceFactory.java:264)
	at io.confluent.ksql.ddl.commands.CreateSourceFactory.createStreamCommand(CreateSourceFactory.java:116)
	at io.confluent.ksql.ddl.commands.CommandFactories.handleCreateStream(CommandFactories.java:129)
	at io.confluent.ksql.util.HandlerMaps$BuilderR2.lambda$castHandler2$2(HandlerMaps.java:840)
	at io.confluent.ksql.ddl.commands.CommandFactories.create(CommandFactories.java:110)
	at io.confluent.ksql.engine.EngineContext.createDdlCommand(EngineContext.java:205)
	at io.confluent.ksql.engine.EngineExecutor.plan(EngineExecutor.java:357)
	... 15 more
``` 

**Additional context**

I am running the [cp-all-in-one](https://github.com/confluentinc/cp-all-in-one/tree/7.0.1-post/cp-all-in-one) Confluent Platform stack in Docker from `7.0.1-post`. I have also tried with the latest `0.24.0` release of KsqlDB.

**Comment by albp-rmt** [comment]
If it helps, I have just re-tested this against the recently released, 7.1.0 confluentinc Docker containers and it worked.

```
$ docker exec -it ksqldb-cli ksql http://ksqldb-server:8088

...
...
...

CLI v7.1.0, Server v7.1.0 located at http://ksqldb-server:8088
Server Status: RUNNING

Having trouble? Type 'help' (case-insensitive) for a rundown of how things work!

ksql> CREATE STREAM s5 (
>    k VARCHAR KEY,
>    ts TIMESTAMP,
>    v1 INT,
>    v2 VARCHAR
>) WITH (
>    kafka_topic = 's5',
>    partitions = 1,
>    value_format = 'avro',
>    timestamp = 'ts'
>);

 Message        
----------------
 Stream created 
----------------
ksql> 

```

**Comment by Gerrrr** [comment]
FWIW, I tried to reproduce the issue in 0.24.0 and the provided example created the stream for me as expected.
